pkg/api/customization/catalog/validator.go

```diff
@@ -0,0 +1,42 @@
+package catalog
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+)
+
+var (
+	controlChars   = regexp.MustCompile("[[:cntrl:]]")
+	controlEncoded = regexp.MustCompile("%[0-1][0-9,a-f,A-F]")
+)
+
+func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	if url, ok := data["url"].(string); ok {
```

Comment:
Let's also return an error if the URL is blank

Reply:
fixed

Reply:
Needed to move this whole validation to POST only, as an update would fail because it doesn't supply URL.